data-prepper-plugins/sqs-source/src/main/java/org/opensearch/dataprepper/plugins/source/sqs/SqsWorker.java

```diff
@@ -0,0 +1,330 @@
+import org.opensearch.dataprepper.model.buffer.Buffer;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+
+public class SqsWorker implements Runnable {
+    private static final Logger LOG = LoggerFactory.getLogger(SqsWorker.class);
+    static final String SQS_MESSAGES_RECEIVED_METRIC_NAME = "sqsMessagesReceived";
+    static final String SQS_MESSAGES_DELETED_METRIC_NAME = "sqsMessagesDeleted";
+    static final String SQS_MESSAGES_FAILED_METRIC_NAME = "sqsMessagesFailed";
+    static final String SQS_MESSAGES_DELETE_FAILED_METRIC_NAME = "sqsMessagesDeleteFailed";
+    static final String SQS_VISIBILITY_TIMEOUT_CHANGED_COUNT_METRIC_NAME = "sqsVisibilityTimeoutChangedCount";
+    static final String SQS_VISIBILITY_TIMEOUT_CHANGE_FAILED_COUNT_METRIC_NAME = "sqsVisibilityTimeoutChangeFailedCount";
+    static final String ACKNOWLEDGEMENT_SET_CALLACK_METRIC_NAME = "acknowledgementSetCallbackCounter";
+
+    private final SqsClient sqsClient;
+    private final SqsEventProcessor sqsEventProcessor;
+    private final Counter sqsMessagesReceivedCounter;
+    private final Counter sqsMessagesDeletedCounter;
+    private final Counter sqsMessagesFailedCounter;
+    private final Counter sqsMessagesDeleteFailedCounter;
+    private final Counter acknowledgementSetCallbackCounter;
+    private final Counter sqsVisibilityTimeoutChangedCount;
+    private final Counter sqsVisibilityTimeoutChangeFailedCount;
+    private final Backoff standardBackoff;
+    private final QueueConfig queueConfig;
+    private int failedAttemptCount;
+    private final boolean endToEndAcknowledgementsEnabled;
+    private final AcknowledgementSetManager acknowledgementSetManager;
+    private volatile boolean isStopped = false;
+    private final BufferAccumulator<Record<Event>> bufferAccumulator;
+    private Map<Message, Integer> messageVisibilityTimesMap;
+
+    public SqsWorker(final Buffer<Record<Event>> buffer,
+                     final AcknowledgementSetManager acknowledgementSetManager,
+                     final SqsClient sqsClient,
+                     final SqsEventProcessor sqsEventProcessor,
+                     final SqsSourceConfig sqsSourceConfig,
+                     final QueueConfig queueConfig,
+                     final PluginMetrics pluginMetrics,
+                     final Backoff backoff) {
+
+        this.sqsClient = sqsClient;
+        this.sqsEventProcessor = sqsEventProcessor;
+        this.queueConfig = queueConfig;
+        this.acknowledgementSetManager = acknowledgementSetManager;
+        this.standardBackoff = backoff;
+        this.endToEndAcknowledgementsEnabled = sqsSourceConfig.getAcknowledgements();
+        this.bufferAccumulator = BufferAccumulator.create(buffer, sqsSourceConfig.getNumberOfRecordsToAccumulate(), sqsSourceConfig.getBufferTimeout());
```

Comment:
Do we need or want a buffer accumulator here? The `Buffer` itself is thread safe. The original `BufferAccumulator` was made to help avoid unnecessary synchronization when writing a large S3 object to the buffer from a single thread.

The way the SQS source works is quite different. Each thread will read up to 10 messages. I'd think those 10 should just be written directly to the `Buffer` after each `ReceiveMessage` call.